tiledb/sm/query/reader.cc

```diff
@@ -1387,41 +1385,55 @@ Status Reader::dense_read() {
 
 template <class T>
 Status Reader::fill_dense_coords(const Subarray& subarray) {
-  // For easy reference
-  uint64_t coords_buff_offset = 0;
-  auto it = attr_buffers_.find(constants::coords);
-  assert(it != attr_buffers_.end());
-  auto coords_buff = it->second.buffer_;
-  auto coords_buff_size = *(it->second.buffer_size_);
+  // Prepare buffers
+  std::vector<unsigned> dim_idx;
+  std::vector<QueryBuffer*> buffers;
+  auto coords_it = buffers_.find(constants::coords);
+  auto dim_num = array_schema_->dim_num();
+  if (coords_it != buffers_.end()) {
+    buffers.emplace_back(&(coords_it->second));
+    dim_idx.emplace_back(dim_num);
+  } else {
+    for (unsigned d = 0; d < dim_num; ++d) {
```

Comment:
buffers.reserve(dim_num);
dim_idx.reserve(dim_num);

Reply:
That will not work, we need to compute the exact size value based on whether coords are needed or not. We can do it of course but it gets a bit complex. Given that `dim_num` here will be very small, the extra cost for buffer alloc here is negligible.